### DISCUSSION TITLE
Is it possible to disable robotidy for file extensions capitalisation?

### DISCUSSION THREAD

**Discussion by piyushasaran12** [MarketSquare/robotframework-tidy#724] [Q&A] [answered]
for e.g if the keyword needs a file as xyz.pdf then after applying robotidy it will be changed to xyz.Pdf and thus test fail because file not found

**Comment by bhirsz** [comment]
Do you have Rename Keywords or RenameVariables enabled? Depending on your config it could be either configurable or bug in Robotidy - would need to check

**Comment by piyushasaran12** [comment]
I have this config in .robotidy

[tool.robotidy]
diff = false
overwrite = true
verbose = false
separator = "space"
spacecount = 4
line_length = 120
lineseparator = "native"
skip_gitignore = false
ignore_git_dir = false
transform =[
    "RenameKeywords:keyword_case=capitalize_words"
]
configure = [
    "AddMissingEnd:enabled=True",
    "NormalizeSeparators:enabled=True",
    "DiscardEmptySections:enabled=True",
    "MergeAndOrderSections:enabled=True",
    "RemoveEmptySettings:enabled=True",
    "ReplaceEmptyValues:enabled=True",
    "NormalizeAssignments:enabled=True",
    "GenerateDocumentation:enabled=False",
    "OrderSettings:enabled=True",
    "OrderSettingsSection:enabled=True",
    "NormalizeTags:enabled=True",
    "OrderTags:enabled=False",
    "RenameVariables:enabled=False",
    "IndentNestedKeywords:enabled=False",
    "AlignSettingsSection:enabled=True",
    "AlignVariablesSection:enabled=True",
    "AlignTemplatedTestCases:enabled=False",
    "AlignTestCasesSection:enabled=False",
    "AlignKeywordsSection:enabled=False",
    "NormalizeNewLines:enabled=True",
    "NormalizeSectionHeaderName:enabled=True",
    "NormalizeSettingName:enabled=True",
    "ReplaceRunKeywordIf:enabled=True",
    "SplitTooLongLine:enabled=True",
    "SmartSortKeywords:enabled=False",
    "RenameTestCases:enabled=False",
    "RenameKeywords:enabled=True",
    "ReplaceReturns:enabled=True",
    "ReplaceBreakContinue:enabled=True",
    "InlineIf:enabled=True",
    "Translate:enabled=False",
    "NormalizeComments:enabled=True",
]

There are 2 problems with robotidy that I face:

1. It changes the email address like it basically change the email address from abc@xyz.com to abc@xyz.Com and this causes issues
2. If the file attachement is abc.pdf or abc.txt, after running robotidy it changes to abc.Pdf and abc.Txt causing issues as file not found.

Please suggest what to do.

**Reply by bhirsz** [reply, marked as answer]
Thanks. Additional question: is this email / file name a 'normal' argument or embedded argument? For example:

```
    # normal
    Input Email And Validate    embedded@arg.com
    # embedded
    Input Email embedded@arg.com And Validate
    
```

It looks like embedded argument - which can't be easily detected by robotidy and when encountered, is considered as 'keyword name'. That with combination of using ``RenameKeywords`` may lead to renaming arguments. 

Possible solutions:

1. Do not use RenameKeywords transformer (you will lose feature of renaming keywords but arguments should not be renamed any longer)
2. Disable robotidy only for particular line where it causes issues:

```
    Input email 'abc@xyz.com' and validate  #  robotidy: off
```

I would recommend trying with 2) first. I will also take a look if I can improve detection mechanism to not rename in such cases.
